Sources/protoc-gen-swift/main.swift

```diff
@@ -241,12 +241,13 @@ struct GeneratorPlugin {
 
   private func auditProtoCVersion(request: Google_Protobuf_Compiler_CodeGeneratorRequest) {
     guard request.hasCompilerVersion else {
-      Stderr.print("WARNING: unknown version of protoc, use 3.2.x or later to ensure JSON support is correct.")
+      // Don't warn if the compiler version is not explicitly set.
       return
     }
-    // 3.2.x is what added the compiler_version, so there is no need to
```

Comment:
This comment explains exactly why it has to be an unset test vs and explicit version check. Old versions of protoc didn't set the field because it didn't exist and there was no way to detect/report these sorta problems.